### PR TITLE
Consolidate error handling on Custom data import tpl

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Consolidate error handling on Custom data import tpl

Before
----------------------------------------
image

image

After
----------------------------------------

UI is the same - but the tpl code is generic

Technical Details
----------------------------------------
During the last round I consolidated all the other non-contact files - but I got stuck on the error handling. I thought there was a less stupid approach but in the end stupid is no worse that how it was & it gets us past it.

Once consolidated at the tpl layer I can change it in one place to consolidate with Contact - which has the extra `sql import` and lack of consolidation there is the reason I can't fix the default / positioning on column headers

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Custom/Import/Form/DataSource.php [hunk 1]
```diff
@@ -57,13 +96,15 @@ protected function getTranslatedEntities(): string {
...
    */
-  public function setDefaultValues() {
+  public function setDefaultValues(): array {
     $config = CRM_Core_Config::singleton();
     $defaults = [
       'contactType' => CRM_Import_Parser::CONTACT_INDIVIDUAL,
       'fieldSeparator' => $config->fieldSeparator,
-      'multipleCustomData' => $this->_id,
+      // Perhaps never used, but permits url passing of the group.
+      $this->_id = CRM_Utils_Request::retrieve('id', 'Positive', $this),
```
Comment (eileenmcnaughton) [inline CRM/Custom/Import/Form/DataSource.php:107]: moved from the next section

File: templates/CRM/Custom/Import/Form/DataSource.tpl [hunk 2]
```diff
@@ -8,70 +8,71 @@
...
 <div class="crm-block crm-form-block crm-import-datasource-form-block">
 
   {* WizardHeader.tpl provides visual display of steps thru the wizard as well as title for current step *}
   {include file="CRM/common/WizardHeader.tpl"}
- {if !$fieldGroups}
-  <div class="messages warning no-popup">
-    {ts}This import screen cannot be used because there are no Multi-value custom data groups.{/ts}
-  </div>
- {/if}
+  {if $errorMessage}
```
Comment (eileenmcnaughton) [inline templates/CRM/Custom/Import/Form/DataSource.tpl:17]: This is the change

File: templates/CRM/Custom/Import/Form/DataSource.tpl [hunk 3]
```diff
@@ -8,70 +8,71 @@
...
+     {if $savedMapping}
+       <tr class="crm-import-uploadfile-form-block-savedMapping">
+         <td class="label">{$form.savedMapping.label}</td>
+         <td><span>{$form.savedMapping.html}</span> </td>
+       </tr>
+       <tr>
+         <td>&nbsp;</td>
+         <td class="description">{ts}Select Saved Mapping, or leave blank to create a new mapping.{/ts}</td>
+       </tr>
+     {/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Custom/Import/Form/DataSource.tpl:74]: if & tr were mis-ordered

**Comment by colemanw** [comment]
Looks good. Tests pass.

**Pull request merged**